### PR TITLE
Remove redundant code in apimachinery/pkg

### PR DESCRIPTION
**Pull request by HecarimV**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
Remove redundant code 

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

### CODE REVIEW

File: staging/src/k8s.io/apimachinery/pkg/labels/selector.go [hunk 1]
```diff
@@ -943,7 +943,7 @@ func ValidatedSelectorFromSet(ls Set) (Selector, error) {
 // A nil and empty Sets are considered equivalent to Everything().
 // It assumes that Set is already validated and doesn't do any validation.
 func SelectorFromValidatedSet(ls Set) Selector {
-	if ls == nil || len(ls) == 0 {
+	if len(ls) == 0 {
```
Comment (MadhavJivrajani) [inline staging/src/k8s.io/apimachinery/pkg/labels/selector.go:946]: `Set` is essentially a map, and the default value of a map is `nil`. This check is valid imo :)

File: staging/src/k8s.io/apimachinery/pkg/labels/selector.go [hunk 2]
```diff
@@ -923,7 +923,7 @@ func SelectorFromSet(ls Set) Selector {
 // nil and empty Sets are considered equivalent to Everything().
 // The Set is validated client-side, which allows to catch errors early.
 func ValidatedSelectorFromSet(ls Set) (Selector, error) {
-	if ls == nil || len(ls) == 0 {
+	if len(ls) == 0 {
```
Comment (MadhavJivrajani) [inline staging/src/k8s.io/apimachinery/pkg/labels/selector.go:926]: Same comment as below

File: staging/src/k8s.io/apimachinery/pkg/labels/selector.go (hunk 1 above)
Comment (MadhavJivrajani) [reply staging/src/k8s.io/apimachinery/pkg/labels/selector.go:946]: Edit: although a `nil` map has Len of 0

Comment (HecarimV) [reply staging/src/k8s.io/apimachinery/pkg/labels/selector.go:946]: yeah, even nil ones, which have a length of zero. So I think It is not necessary to check nil.

Comment (HecarimV) [reply staging/src/k8s.io/apimachinery/pkg/labels/selector.go:946]: @MadhavJivrajani Thanks for your review, should I revert here?

**Pull request closed without merging**